packages/jest-haste-map/src/__tests__/fastpath-test.js

```diff
@@ -0,0 +1,575 @@
+      process.platform = actualPlatform;
+
+      fp = require('../fastpath');
+      expect(fp.extname(__filename)).toEqual('.js');
+      expect(fp.extname('')).toEqual('');
+      expect(fp.extname('/path/to/file')).toEqual('');
+      expect(fp.extname('/path/to/file.ext')).toEqual('.ext');
+      expect(fp.extname('/path.to/file.ext')).toEqual('.ext');
+      expect(fp.extname('/path.to/file')).toEqual('');
+      expect(fp.extname('/path.to/.file')).toEqual('');
+      expect(fp.extname('/path.to/.file.ext')).toEqual('.ext');
+      expect(fp.extname('/path/to/f.ext')).toEqual('.ext');
+      expect(fp.extname('/path/to/..ext')).toEqual('.ext');
+      expect(fp.extname('file')).toEqual('');
+      expect(fp.extname('file.ext')).toEqual('.ext');
+      expect(fp.extname('.file')).toEqual('');
+      expect(fp.extname('.file.ext')).toEqual('.ext');
+      expect(fp.extname('/file')).toEqual('');
+      expect(fp.extname('/file.ext')).toEqual('.ext');
+      expect(fp.extname('/.file')).toEqual('');
+      expect(fp.extname('/.file.ext')).toEqual('.ext');
+      expect(fp.extname('.path/file.ext')).toEqual('.ext');
+      expect(fp.extname('file.ext.ext')).toEqual('.ext');
+      expect(fp.extname('file.')).toEqual('.');
+      expect(fp.extname('.')).toEqual('');
+      expect(fp.extname('./')).toEqual('');
+      expect(fp.extname('.file.ext')).toEqual('.ext');
+      expect(fp.extname('.file')).toEqual('');
+      expect(fp.extname('.file.')).toEqual('.');
+      expect(fp.extname('.file..')).toEqual('.');
+      expect(fp.extname('..')).toEqual('');
+      expect(fp.extname('../')).toEqual('');
+      expect(fp.extname('..file.ext')).toEqual('.ext');
+      expect(fp.extname('..file')).toEqual('.file');
+      expect(fp.extname('..file.')).toEqual('.');
+      expect(fp.extname('..file..')).toEqual('.');
+      expect(fp.extname('...')).toEqual('.');
+      expect(fp.extname('...ext')).toEqual('.ext');
+      expect(fp.extname('....')).toEqual('.');
+      expect(fp.extname('file.ext/')).toEqual('.ext');
+      expect(fp.extname('file.ext//')).toEqual('.ext');
+      expect(fp.extname('file/')).toEqual('');
+      expect(fp.extname('file//')).toEqual('');
+      expect(fp.extname('file./')).toEqual('.');
+      expect(fp.extname('file.//')).toEqual('.');
+    });
+
+    it('should handle path backslashes (win32)', () => {
+      process.platform = 'win32';
+
+      fp = require('../fastpath');
+      // On windows, backspace is a path separator.
+      expect(fp.extname('.\\')).toEqual('');
+      expect(fp.extname('..\\')).toEqual('');
+      expect(fp.extname('file.ext\\')).toEqual('.ext');
+      expect(fp.extname('file.ext\\\\')).toEqual('.ext');
+      expect(fp.extname('file\\')).toEqual('');
+      expect(fp.extname('file\\\\')).toEqual('');
+      expect(fp.extname('file.\\')).toEqual('.');
+      expect(fp.extname('file.\\\\')).toEqual('.');
+    });
+
+    it('should handle path backslashes (posix)', () => {
```

Comment:
don’t we need to set `process.platform = 'linux'` here to make these tests pass on windows?